### PR TITLE
Call hearShake() only once for each shake.

### PR DESCRIPTION
**Pull request by arnt**
Until now, shaking continuously for a long time (say, a second and a
half) could cause hearShake() to be called several times. This change
ensures that there is a period of immobility after the hearShake() call
before the next shake can be recognized.

### CODE REVIEW

File: library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java [hunk 1]
```diff
@@ -71,4 +73,102 @@ private void assertContent(ShakeDetector.SampleQueue q, boolean... expected) {
...
+
+  @Test public void testDisregardShortShakes() {
+    ShakeCounter c = new ShakeCounter();
+    ShakeDetector s = new ShakeDetector(c);
+
+    addShake(s, 200);
+    addSilence(s, 1000);
+    addShake(s, 200);
+
+    assertThat(c.shakes == 0).isTrue();
```
Comment (JakeWharton) [inline library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java]:
```java
assertThat(c.shakes).isZero();
```

File: library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java [hunk 2]
```diff
@@ -71,4 +73,102 @@ private void assertContent(ShakeDetector.SampleQueue q, boolean... expected) {
...
+    ShakeDetector s = new ShakeDetector(c);
+
+    addShake(s, 1000);
+    addSilence(s, 100);
+    addShake(s, 1000);
+    assertThat(c.shakes == 1).isTrue();
+
+    addSilence(s, 1000);
+    addShake(s, 1000);
+    assertThat(c.shakes == 2).isTrue();
```
Comment (JakeWharton) [inline library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java]:
```java
assertThat(c.shakes).isEqualTo(2);
```

File: library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java [hunk 3]
```diff
@@ -71,4 +73,102 @@ private void assertContent(ShakeDetector.SampleQueue q, boolean... expected) {
...
+   */
+
+  @Test public void testDisregardCloseShakes() {
+    ShakeCounter c = new ShakeCounter();
+    ShakeDetector s = new ShakeDetector(c);
+
+    addShake(s, 1000);
+    addSilence(s, 100);
+    addShake(s, 1000);
+    assertThat(c.shakes == 1).isTrue();
```
Comment (JakeWharton) [inline library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java]:
```java
assertThat(c.shakes).isEqualTo(1);
```

File: library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java [hunk 4]
```diff
@@ -71,4 +73,102 @@ private void assertContent(ShakeDetector.SampleQueue q, boolean... expected) {
...
+
+  @Test public void testHearTwoShakes() {
+    ShakeCounter c = new ShakeCounter();
+    ShakeDetector s = new ShakeDetector(c);
+
+    addShake(s, 1000);
+    addSilence(s, 1000);
+    addShake(s, 1000);
+
+    assertThat(c.shakes == 2).isTrue();
```
Comment (JakeWharton) [inline library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java]:
```java
assertThat(c.shakes).isEqualTo(2);
```

File: library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java [hunk 5]
```diff
@@ -71,4 +73,102 @@ private void assertContent(ShakeDetector.SampleQueue q, boolean... expected) {
...
+
+
+  /** Tests that a one-second shake results in one hearShake() call. */
+  @Test public void testHearShake() {
+    ShakeCounter c = new ShakeCounter();
+    ShakeDetector s = new ShakeDetector(c);
+
+    addShake(s, 1000);
+
+    assertThat(c.shakes == 1).isTrue();
```
Comment (JakeWharton) [inline library/src/test/java/com/squareup/seismic/ShakeDetectorTest.java]:
```java
assertThat(c.shakes).isEqualTo(1);
```

**Comment by JakeWharton** [comment]
I'll just fix it up locally and merge.

**Comment by arnt** [comment]
Thanks, because I'm not too happy about doing such changes in test code.

In tests I'll cut-and-paste, use suboptimal functions, whatever so long as I'm increasing test coverage. Given a choice between two minutes spent prettifying tests and increasing test coverage, I'm almost always going to pick the latter.

**Comment by JakeWharton** [comment]
Test code is just code and it must adhere to style rules.

**Comment by arnt** [comment]
OK.

You might want to look at the detector itself; the physics model currently doesn't take account of earth's gravity. I have a patch FWIW.

**Pull request closed without merging**